src/Stratis.SmartContracts.Executor.Reflection/MethodCall.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+
+namespace Stratis.SmartContracts.Executor.Reflection
+{
+    /// <summary>
+    /// Represents a call to a contract method.
+    /// </summary>
+    public class MethodCall
+    {
+        private readonly string methodName;
+
+        public const string FallbackMethodName = nameof(SmartContract.Fallback);
+
+        public const string ExternalFallbackMethodName = "";
+
+        public MethodCall(string methodName, object[] methodParameters = null)
+        {
+            this.methodName = methodName;
+            this.Parameters = methodParameters;
+        }
+
+        public static MethodCall Fallback()
+        {
+            return new MethodCall(ExternalFallbackMethodName);
+        }
+
+        public object[] Parameters { get; }
+
+        public string Name
+        {
+            get
+            {
+                return this.IsFallbackCall
+                    ? FallbackMethodName
+                    : this.methodName;
+            }
+        }
+
+        /// <summary>
+        /// Returns true if the method call is a fallback method call.
+        /// </summary>
+        public bool IsFallbackCall
+        {
+            get
+            {
+                // The fallback method must always be the override with no parameters,
+                // so it's not enough just to check if the method name is correct.
+                return (this.Parameters == null || this.Parameters.Length == 0) 
```

Comment:
I believe this means that a call to "" that has parameters is therefore not going to be considered a fallback call. 

I wonder what we decide here?